eisbuk-admin/src/components/atoms/OrganizationSettings/OrganizationSettings.tsx

```diff
@@ -0,0 +1,397 @@
+import React, { useState } from "react";
+import * as Yup from "yup";
+import { OrganizationData } from "eisbuk-shared/dist";
+
+import { Formik, Field, Form, useField } from "formik";
+import { ValidationMessage } from "@/enums/translations";
+
+import FormControl from "@material-ui/core/FormControl";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import Cancel from "@material-ui/icons/Cancel";
+
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+import { Divider, IconButton, Typography } from "@material-ui/core";
+import clsx from "clsx";
+import { useDispatch, useSelector } from "react-redux";
+import { getLocalAuth } from "@/store/selectors/auth";
+import { updateOrganization } from "@/store/actions/organizationOperations";
+import i18n from "@/i18next/i18n";
+import ErrorMessage from "@/components/atoms/ErrorMessage";
+
+// #region validations
+const OrganizationValidation = Yup.object().shape({
+  smsFrom: Yup.string()
+    .matches(/^[a-z0-9]+$/, i18n.t(ValidationMessage.InvalidSmsFrom))
+    .max(11, i18n.t(ValidationMessage.InvalidSmsFromLength)),
+});
+// #endregion validations
+interface Props {
+  organization: OrganizationData;
+}
+
+interface FieldProps {
+  name: string;
+  label: string;
+}
+const OrganizationSettings: React.FC<Props> = ({ organization }) => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  const userAuthInfo = useSelector(getLocalAuth);
+
+  const [enableEdit, setEnableEdit] = useState(false);
+
+  const handleSubmit = (orgData: OrganizationData) => {
+    dispatch(updateOrganization(orgData, organization.organizationName || ""));
+    setEnableEdit(false);
+  };
+
+  const handleEnableEdit = (resetForm: () => void) => {
+    if (enableEdit) resetForm();
+    setEnableEdit(!enableEdit);
+  };
+
+  const emailFields: Partial<FieldProps>[] = [
```

Comment:
I would maybe make a type alias as you would use it throughout: for both `emailFields` and `smsFields` as well as mapping function declaration.

Reply:
Actually, I don't remember why I used `Partial` but I don't need to anymore so I'll just leave it as `FieldProps[]`